File: test/fixtures/orphanhelp/android/src/main/java/com/example/orphanhelp/OrphanHelpPlugin.java
package com.example.orphanhelp;

import com.getcapacitor.JSObject;
import com.getcapacitor.PermissionCallback;
import com.getcapacitor.Plugin;
import com.getcapacitor.PluginCall;
import com.getcapacitor.PluginMethod;
import com.getcapacitor.annotation.CapacitorPlugin;

@CapacitorPlugin(name = "OrphanHelp")
public class OrphanHelpPlugin extends Plugin {

    private Greeter greeter;

    @PluginMethod
    public void ping(final PluginCall call) {
        JSObject ret = new JSObject();
        ret.put("pong", true);
        call.resolve(ret);
    }

    @PluginMethod
    public void notHere(PluginCall call) {
        call.unimplemented();
    }

    @PluginMethod
    public void greet(PluginCall call) {
        String who = call.getString("who");
        greeter.greet(who);
        call.resolve();
    }

    void onGreetingEvent(String which) {
        JSObject data = new JSObject();
        data.put("which", Greeter.GREETING_HI);
        notifyListeners("greeted", data);
        bridge.triggerWindowJSEvent("greeted", data);
    }

    @PermissionCallback
    private void permsDone(PluginCall call) {
        call.resolve();
    }
}

File: test/fixtures/orphanhelp/ios/Sources/OrphanHelpPlugin.swift
import Foundation
import Capacitor

@objc(OrphanHelpPlugin)
public class OrphanHelpPlugin: CAPPlugin, CAPBridgedPlugin {
    public let identifier = "OrphanHelpPlugin"
    public let jsName = "OrphanHelp"
    public let pluginMethods: [CAPPluginMethod] = []

    @objc func ping(_ call: CAPPluginCall) {
        let ret = JSObject()
        ret["pong"] = true
        call.resolve(ret)
    }

    @objc func notHere(_ call: CAPPluginCall) {
        call.unimplemented()
    }

    @objc func greet(_ call: CAPPluginCall) {
        call.resolve()
    }
}
